Enhance DDP docstrings for DDP + RPC support.

pritamdamania87 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40139 Add unit test to verify DDP + RPC correctness.
* #40130 Remove `finalize_bucket_sparse` from DDP
* **#39916 Enhance DDP docstrings for DDP + RPC support.**
* #39909 Avoid using default process group in ProcessGroupAgent.
![Screen Shot 2020-06-11 at 7 11 51 PM](https://user-images.githubusercontent.com/9958665/84457511-b76fbc80-ac17-11ea-9615-25147d2deed1.png)

Differential Revision: [D22013190](https://our.internmc.facebook.com/intern/diff/D22013190/)

pritamdamania committed:
Enhance DDP docstrings for DDP + RPC support.



Differential Revision: [D22013190](https://our.internmc.facebook.com/intern/diff/D22013190/)

[ghstack-poisoned]

rohan-varma commented on torch/nn/parallel/distributed.py:
```diff
@@ -148,6 +148,28 @@ class DistributedDataParallel(Module):
         0, to all other replicas in the system in every iteration.
 
+    .. note::
+        If you are using DistributedDataParallel in conjuction with the
```
Should we link to `DistributedDataParallel` doc?

rohan-varma commented on torch/nn/parallel/distributed.py:
```diff
@@ -148,6 +148,28 @@ class DistributedDataParallel(Module):
+        >>>     dist_autograd.backward(context_id, loss)
+
+    .. warning::
+        Using DistributedDataParallel in conjuction with the
```
Here too

rohan-varma commented:
Is there anything that the user needs to be aware of when using `DistributedOptimizer` with DDP + RPC or should that work out of the box?

pritamdamania committed:
Update on "Enhance DDP docstrings for DDP + RPC support."

![Screen Shot 2020-06-11 at 7 11 51 PM](https://user-images.githubusercontent.com/9958665/84457511-b76fbc80-ac17-11ea-9615-25147d2deed1.png)

Differential Revision: [D22013190](https://our.internmc.facebook.com/intern/diff/D22013190/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Enhance DDP docstrings for DDP + RPC support."

![Screen Shot 2020-06-11 at 7 11 51 PM](https://user-images.githubusercontent.com/9958665/84457511-b76fbc80-ac17-11ea-9615-25147d2deed1.png)

Differential Revision: [D22013190](https://our.internmc.facebook.com/intern/diff/D22013190/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Enhance DDP docstrings for DDP + RPC support."

![Screen Shot 2020-06-11 at 7 11 51 PM](https://user-images.githubusercontent.com/9958665/84457511-b76fbc80-ac17-11ea-9615-25147d2deed1.png)

Differential Revision: [D22013190](https://our.internmc.facebook.com/intern/diff/D22013190/)

[ghstack-poisoned]

pritamdamania committed:
Update on "Enhance DDP docstrings for DDP + RPC support."

![Screen Shot 2020-06-11 at 7 11 51 PM](https://user-images.githubusercontent.com/9958665/84457511-b76fbc80-ac17-11ea-9615-25147d2deed1.png)

Differential Revision: [D22013190](https://our.internmc.facebook.com/intern/diff/D22013190/)

[ghstack-poisoned]
